apps/idos-enclave/src/lib/enclave.js

```diff
@@ -39,12 +39,15 @@ export class Enclave {
     };
   }
 
-  async keys(authMethod) {
-    if (authMethod) await this.#openDialog("auth");
-    await this.ensurePassword();
-    await this.ensureKeyPair();
-
-    return this.keyPair.publicKey;
+  async keys() {
+    try {
+      await this.ensurePassword();
+      await this.ensureKeyPair();
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      return this.keyPair?.publicKey;
+    }
```

Comment:
This finally feels weird. I'd expect something like:
```suggestion
    }
    return this.keyPair?.publicKey;
```